geopandas/geoseries.py

```diff
@@ -53,6 +53,37 @@ def _geoseries_constructor_with_fallback(data=None, index=None, crs=None, **kwar
         return Series(data=data, index=index, **kwargs)
 
 
+def _geoseries_expanddim(data=None, index=None, crs=None, **kwargs):
+    from geopandas import GeoDataFrame
+
+    # DataFrame == pd.Series._constructor_expanddim
+    df = pd.DataFrame(data=data, index=index, **kwargs)
```

Comment:
Note there are non trivial changes here in the last new commit: https://github.com/geopandas/geopandas/pull/2296/commits/4cb284d08b4e303d584d476ac5acdf855dde29ff.
I know that's probably annoying, but I think this is much safer with not needing to deal with pandas internals.

Reply:
(This still had a blockmanager case to deal with so perhaps not any better)

Reply:
Since the "upstream" version of this is simply `pd.DataFrame`, I think it totally makes sense to just start with calling this here as well (that's similar as how the GeoDataFrame constructor starts with `super().__init__(data, *args, **kwargs)`)